### DISCUSSION TITLE
Running Scripts on drive insertion/mount

### DISCUSSION THREAD

**Discussion by BillyCroan** [storaged-project/udisks#1078] [General]
I found udisksctl monitor in a recent project where I want to automate the running of scripts when specific usb sticks or memory cards are connected.

I want to be very sure it's the right one.  checking:
serial number from the block device
UUID from the partition header
UUID from the filesystem
Volume Label from the filesystem
and perhaps presence of an identifying file in the filesystem

And only run the script if everything matches.

udisksctl monitor outputs multiple lines during one 'event' which makes it tricky to process in a bash read loop.

Is there a way to get one long line per event? Or Maybe invoke a script for each event, with each line's key-value pair as an environment variable, accessible to that script?

**Comment by vojtechtrefny** [comment]
Parsing the `udisksctl monitor` output probably isn't the easiest way to get the information. UDisks is primarly a DBus daemon, so you could try to use the DBus signals emitted when a new device is added. But even that would be relatively hard with just shell and something like `dbus-monitor`, you'd need to use some scripting language with a good DBus library, like Python. The easiest way might be to go one level lower and use `udevadm monitor` which has simpler output so with `udevadm monitor -u -s "block"` you'll get something like

```
UDEV  [21822.199594] add      /devices/pci0000:00/0000:00:14.0/usb2/2-4/2-4:1.0/host0/target0:0:0/0:0:0:0/block/sda (block)
UDEV  [21822.318745] add      /devices/pci0000:00/0000:00:14.0/usb2/2-4/2-4:1.0/host0/target0:0:0/0:0:0:0/block/sda/sda3 (block)
UDEV  [21822.380798] add      /devices/pci0000:00/0000:00:14.0/usb2/2-4/2-4:1.0/host0/target0:0:0/0:0:0:0/block/sda/sda1 (block)
UDEV  [21822.428106] add      /devices/pci0000:00/0000:00:14.0/usb2/2-4/2-4:1.0/host0/target0:0:0/0:0:0:0/block/sda/sda2 (block)
```

after connecting a flash drive with 3 partitions and then you can get detailed information with `udevadm info <device>` which has also some advanced query options:

```
$ udevadm info --query=property --property="DEVNAME,ID_FS_UUID,ID_FS_LABEL" /sys/devices/pci0000:00/0000:00:14.0/usb2/2-4/2-4:1.0/host0/target0:0:0/0:0:0:0/block/sda/sda2
DEVNAME=/dev/sda2
ID_FS_LABEL=ANACONDA
ID_FS_UUID=1755-D377
```
